src/controllers/convert.controller.ts

```diff
@@ -0,0 +1,65 @@
+import { NextFunction, Request, Response, Router } from 'express';
+import YAML from 'js-yaml';
+
+import { Controller, ConvertRequestDto } from '../interfaces';
+
+import { documentValidationMiddleware } from '../middlewares/document-validation.middleware';
+
+import { ConvertService } from '../services/convert.service';
+
+import { ProblemException } from '../exceptions/problem.exception';
+import { parse, prepareParserConfig } from '../utils/parser';
+
+/**
+ * Controller which exposes the Convert functionality
+ */
+export class ConvertController implements Controller {
+  public basepath = '/convert';
+
+  private convertService = new ConvertService();
+
+  private async convert(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { version, language, asyncapi } = req.body as ConvertRequestDto;
+
+      await parse(asyncapi, prepareParserConfig(req));
+      const convertedSpec = await this.convertService.convertSpec(
+        asyncapi,
+        language,
+        version.toString(),
+      );
+
+      if (!convertedSpec) {
+        return next(new ProblemException({
+          type: 'invalid-json',
+          title: 'Bad Request',
+          status: 400,
+          detail: 'Couldn\'t convert the spec to the requested version.'
+        }));
+      }
+      const convertedSpecObject = YAML.load(convertedSpec);
+      res.json({
+        asyncapi: convertedSpecObject
+      });
```

Comment:
I know that it's a secret knowledge, because we don't have a deep dive docs for our ParserJS, but ParserJS throws error when you pass to it the old version (1.0.0, 1.1.0, 1.2.0 and all 2.0.0 rcs) - here is a line https://github.com/asyncapi/parser-js/blob/master/lib/parser.js#L79

but converter should allow to make operation on that versions (convert it to newest), so I think that we have problem here. We should "filter" old version and even if `parse` will throw error, we should perform ahead actions. I hope that you understand.

And last thing:

```js
const convertedSpecObject = YAML.load(convertedSpec);
```

as I described in the previous comments, we have such a logic with converting to the YAML/JSON in the converter-js side, but we can elaborate on that topic :)